File: java/src/HelloBranch.java
/**
 * Say hello.
 * @author RyoWakabayashi
 * @version 1.0
 */
public final class HelloBranch {
  private HelloBranch() {
    throw new IllegalStateException("Utility class");
  }

  /**
    * main.
    * @param args Dummy
    */
  public static void main(final String[] args) {
    String helloWorld = "Hello, world!";

    if (helloWorld == "Good morning") {
      System.out.println("morning");
    } else if (helloWorld.contains("Hello")) {
      System.out.println("noon");
    } else {
      System.out.println("night");
    }

    switch (helloWorld) {
      case "Hello, world!":
        System.out.println("nonn");
        break;
      case "Good morning, world!":
        System.out.println("morning");
        break;
      default:
        System.out.println("night");
    }
  }
}

File: golang/src/hello_branch.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	helloWorld := "Hello, world!"

	switch {
	case helloWorld == "Good morning":
		fmt.Println("morning")
	case strings.Contains(helloWorld, "Hello"):
		fmt.Println("noon")
	default:
		fmt.Println("night")
	}

	switch helloWorld {
	case "Hello, world!":
		fmt.Println("noon")
	case "Good morning, world!":
		fmt.Println("morning")
	default:
		fmt.Println("night")
	}
}
